test/core/base/check.js

```diff
@@ -98,6 +98,62 @@ describe('Check', function() {
 				assert.equal('fong', check.test());
 				delete Check.prototype.test;
 			});
+			it('should override evaluate as ID', function() {
+				axe._load({});
+				axe._audit.metadataFunctionMap['custom-evaluate'] = function() {
+					return 'fong';
+				};
+
+				Check.prototype.test = function() {
+					return this.evaluate();
+				};
+				var check = new Check({
+					evaluate: 'function () { return "foo"; }'
```

Comment:
What happens if we do this:

```js
evaluate: '() => "foo"'
```

Reply:
As far as I know we don't allow that. Although it is possible, we only support doT.js template output which uses `functions`.

@WilcoFiers do you find this a problem (especially since we deprecated this?)